sheepdog/utils/transforms/graph_to_doc.py

```diff
@@ -766,6 +673,75 @@ def validate_export_node(node_label):
         raise UserError("cannot export node with category `internal`")
 
 
+def format_prop(prop):
+    """
+    Map over ``titles`` to get properties usable for looking up from
+    node instances.
+
+    Change properties to have 'node_id' instead of 'id', and 'label'
+    instead of 'type', so that the props can be looked up as dictionary
+    entries:
+
+    .. code-block:: python
+
+        node[prop]
+
+    """
+    if prop == "id":
+        return "node_id"
+    elif prop == "type":
+        return "label"
+    return prop
+
+
+def format_linked_prop(prop):
+    """
+    For properties which link to other nodes, convert the link to a
+    tuple such that the linked property can be looked up using the
+    elements of the tuple as indexes. For example, for a ``Case`` node
+    with a link to `experiments.id`:
+
+    .. code-block:: python
+
+        # prop == ('experiments', 'node_id')
+        node[prop[0]][prop[1]][prop[2]]
+    """
+    link_name, link_alias = split_link(prop)
+    return (link_name, format_prop(link_alias))
+
+
+def get_all_titles(node_label):
+    # Get the template for this node, which is basically the column
+    # headers in the resulting TSV.
+    # Example ``titles_non_linked``:
+    #     [
+    #         'type', 'id', 'submitter_id', 'disease_type', 'primary_site'
+    #     ]
+    # Example ``titles_linked``:
+    #     [
+    #         'experiments.id', 'experiments.submitter_id'
+    #     ]
+
+    unstripped_template = entity_to_template(
+        node_label, exclude_id=False, file_format="tsv"
+    )
+    # Strip asterisks.
+    template = [prop.lstrip("*") for prop in unstripped_template]
+    # Get the titles so the linked fields are at the end (to match the
+    # structure of the query we will run later).
+    titles_non_linked = []
+    titles_linked = []
+    for title in template:
+        if is_link_field(title):
+            titles_linked.append(title)
+        else:
+            # 'urls' is part of the templates but not part of the dicts
+            # and not exported, so we remove it here
+            if title != "urls":
+                titles_non_linked.append(title)
+    return titles_non_linked, titles_linked
+
+
 def export_all(node_label, project_id, file_format, db, **kwargs):
```

Comment:
TSV export still returns "," instead of an empty string when there is no link
JSON export still returns empty links too, for example I got `"core_metadata_collections": [{"node_id": "", "submitter_id": ""}`